tests/rptest/clients/kafka_cli_tools.py

```diff
@@ -274,38 +277,25 @@ def reassign_partitions(self,
         else:
             raise NotImplementedError(f"Unknown operation: {operation}")
 
-        return self._run("kafka-reassign-partitions.sh", args)
-
-    def execute_reassign_partitions(self, reassignments: dict):
-        output = self.reassign_partitions(reassignments=reassignments,
-                                          operation="execute")
-        return output.splitlines()
-
-    def verify_reassign_partitions(self, reassignments: dict, timeout_s: int):
         output = None
 
         def do_reassign_partitions():
             nonlocal output
-            output = self.reassign_partitions(
-                reassignments=reassignments,
-                operation="verify",
-                write_reassignments_to_disk=False)
-
-            # Retry the script if there is reassignment still in progress
-            return "is still in progress." not in output
+            try:
+                output = self._run("kafka-reassign-partitions.sh", args)
+            except subprocess.CalledProcessError as e:
+                self.logger.debug(f"Error {e.returncode}, output {e.output}")
+                # Retry on JsonParseException
+                if "JsonParseException: The input string shouldn't be empty" in e.output:
+                    return False
```

Comment:
are you sure this is an issue with the Kafka tool? I see that the referenced issue failed in the context of cancelling partitions:

```
  File "/root/tests/rptest/clients/kafka_cli_tools.py", line 314, in cancel_reassign_partitions
    output = self.reassign_partitions(reassignments=reassignments,
```

but looking at the cancel helper I see

```
write_reassignments_to_disk=False,
```

and from what I can tell the json file won't be written in that case so it is expected to be empty/not exist?

Reply:
> are you sure this is an issue with the Kafka tool? I see that the referenced issue failed in the context of cancelling partitions:
> 
> ```
>   File "/root/tests/rptest/clients/kafka_cli_tools.py", line 314, in cancel_reassign_partitions
>     output = self.reassign_partitions(reassignments=reassignments,
> ```
> 
> but looking at the cancel helper I see
> 
> ```
> write_reassignments_to_disk=False,
> ```
> 
> and from what I can tell the json file won't be written in that case so it is expected to be empty/not exist?


Yes I'm sure. I've seen this failure in CI with `write_reassignments_to_disk=True` as well. It happened last night in an older run of #8849 

Nevertheless, I thought I changed that param to be `write_reassignments_to_disk=True`. I'll edit and push.

Reply:
I'll get back to this PR after #8849 is merged as it is a fix for a higher severity issue.

Reply:
Did you consider that the issue may be because multiple partition reassignment tests are running in parallel within the same ducktape runner and may be clobbering each other when they each call `open("reassignments.json", "w")` which will truncate the file before writing?

Reply:
Good point. I may look into using a python Tempfile for this then so each runner has a unique json filename.

Reply:
good idea. you'll want to use NamedTemporaryFile with the delete=False option. Once you write your data into the file make sure to flush/close the file, then use `f.name` to pass the path of the temp file to the Kafka tool. Finally, you can wrap it all in a try/finally block to remove the temp file.